test/e2e/istio_test.go

```diff
@@ -364,6 +382,26 @@ func getKialiWorkloadHealth(t *testing.T, kialiAPIUrl string, namespace, workloa
 	return &health, nil
 }
 
+// getKialiWorkloadHealthIstioByWorkloadEndpoint gets metrics of workload by /namespaces/<ns>/workloads/<workload> API.
+// Used in Istio 1.18 and later. Istio 1.17 does not have this API.
+func getKialiWorkloadHealthIstioByWorkloadEndpoint(t *testing.T, kialiAPIUrl string, namespace, workloadName string) (*workloadHealth, error) {
+	t.Helper()
+
+	kialiWorkloadURL := fmt.Sprintf("%s/namespaces/%s/workloads/%s", kialiAPIUrl, namespace, workloadName)
+	resp, err := helpers.DefaultHTTPClient().Get(kialiWorkloadURL)
+	require.NoErrorf(t, err, "failed to call Kiali workload API %s", kialiWorkloadURL)
+	defer resp.Body.Close()
+	// Verify the workload response.
+	require.Equalf(t, http.StatusOK, resp.StatusCode, "Got code %d from Kiali workload API %s", resp.StatusCode, kialiWorkloadURL)
+	b, err := io.ReadAll(resp.Body)
+	require.NoError(t, err, "failed to read from response from Kiali workload API")
+	status := workloadStatus{}
+	if err := json.Unmarshal(b, &status); err != nil {
+		return nil, err
+	}
```

Comment:
```suggestion
require.NoError(t, json.Unmarshal(b, &status), "failed to parse response from workload API")
```

So probably this could be replaced with

Reply:
Changing this to `require` seems to help us to acknowledge the API changes, so I'll change that.